### PR TITLE
Remove check_includes = false for //brave/browser/extensions 

### PR DESCRIPTION
**Pull request by bridiver**
Resolves https://github.com/brave/brave-browser/issues/10661
Resolves https://github.com/brave/brave-browser/issues/41398

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/extensions/BUILD.gn [hunk 1]
```diff
@@ -15,14 +17,110 @@ import("//components/gcm_driver/config.gni")
...
+    "//components/content_settings/core/common",
+    "//url",
+  ]
+}
+
+# This is split out from extensions because it has dependencies on
+# //chrome/browser that can't easily be resolved. This is a temporary solution.
+source_set("brave_rewards_api") {
+  # Remove when https://github.com/brave/brave-browser/issues/10661 is resolved
+  check_includes = false
```
[the remark this answers is not in the document]
Comment (bridiver) [reply browser/extensions/BUILD.gn:93]: This is minimizing the scope of `check_includes = false`
@thypon the diff actually caused semgrep to miss a new one that was added above that used to be in `//brave/browser/extensions`, but is now in `brave_prefs_util_impl`

File: browser/tor/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "//components/prefs",
+    "//components/safe_browsing/core/common:safe_browsing_prefs",
+    "//components/user_prefs",
+    "//content/public/browser",
+    "//url",
+  ]
+}
+
+source_set("util") {
+  check_includes = false  # workaround for incognito mode prefs
```
[the remark this answers is not in the document]
Comment (bridiver) [reply browser/tor/BUILD.gn:55]: There was no practical way to put `browser/tor` in its own BUILD.gn without this right now :(

**Review by tmancey — APPROVED** [review]
ads LGTM

**Review by emerick — APPROVED** [review]
Rewards ++

**Review by zenparsing — APPROVED** [review]

File: browser/extensions/BUILD.gn [hunk 3]
```diff
@@ -15,14 +17,116 @@ import("//components/gcm_driver/config.gni")
...
+  ]
+}
+
+# This is split out from extensions because it has dependencies on
+# //chrome/browser/ui that can't easily be resolved. This is a temporary
+# workaround.
+source_set("brave_rewards_api") {
+  # Remove and combine with extensions when
+  # https://github.com/brave/brave-browser/issues/41419 is resolved
+  check_includes = false
```
[the remark this answers is not in the document]
Comment (bridiver) [reply browser/extensions/BUILD.gn:98]: same as above

**Comment by bridiver** [comment]
There are no functional changes and this fixes an intermittent build error so I'm going to admin merge

**Review by darkdh — COMMENTED** [review]
++

**Pull request merged**